[Lee Yong Ning] iP

yongning0310 opened the pull request:
## Dong - Free Your Mind from Tasks!

> _“To Do or Not to Do, That Is the Question” – Unknown 

**Dong** helps you offload the burden of remembering every task. Why? Because:

- It's **text-based**.
- It's **easy to learn**.
- It's **FAST. SUPER FAST** to use.

### How to use?
1. **Download** it from [here](#).
2. **Double-click** to launch.
3. **Add** your tasks.
4. Let **Dong** manage them for you 😉.

### Oh, and it's FREE!

### Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

📌 For all the Java programmers out there, you can use **Dong** to practice your Java skills. Check out the main method:

```java
    public static void main(String[] args) {
        new Duke("./data/tasks.txt").run();
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

yongning0310 committed:
changed name to dong

yongning0310 committed:
added scanner to echo input

yongning0310 committed:
keeps track of list and prints out when required

yongning0310 committed:
created new class for task

yongning0310 committed:
automated testing

yongning0310 committed:
handle exception

yongning0310 committed:
delete task

yongning0310 committed:
edited test cases

yongning0310 committed:
java docs updated

yongning0310 committed:
implemented duke.txt to store db

yongning0310 committed:
added dateTimeUtils to handle dateFormat

yongning0310 committed:
more OOP

yongning0310 committed:
A-Packages

yongning0310 committed:
merged with gradle

yongning0310 committed:
set up gradle

yongning0310 committed:
J unit tests

yongning0310 committed:
built Duke.jar

yongning0310 committed:
java docs, coding standard

yongning0310 committed:
added checkstyle and fixed styling

yongning0310 committed:
added GUI

syamfarh commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,194 @@
+            addToOutput(ui.printTaskContainingKeyword(taskList, findDescription.substring(5).trim()), output);
+            break;
+        default:
+            throw new InvalidInputException();
```
I think you missed a break here

syamfarh commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,91 @@
+                        break;
+                    default:
+                        System.out.println("Unexpected task type encountered: " + line);
+                    }
```
i think you missed a break here as well

syamfarh commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,91 @@
+
+        try {
+            Path path = Paths.get(this.filePath);
+            Files.write(path, content.toString().getBytes());
```
does the writer object need to be closed after writing to file? I'm not sure if Files class need to be closed after used

syamfarh commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,110 @@
+                + "█  █  ▀████ █ █  █ █   █ \n"
+                + "███▀        █  █ █  ███  \n"
+                + "            █   ██       \n"
+                + "                         ";
```
very nice logo!

syamfarh reviewed (commented):
LGTM!

miljyy commented on src/main/java/duke/DialogBox.java:
```diff
@@ -1,47 +1,61 @@
 import javafx.scene.image.ImageView;
 import javafx.scene.layout.HBox;
 
+/**
+ * An example of a custom control using FXML.
+ * This control represents a dialog box consisting of an ImageView to represent the speaker's face and a label
+ * containing text from the speaker.
+ */
```
I like how detailed and clear the Javadoc for this class is!

miljyy commented on src/main/java/duke/taskclasses/TaskList.java:
```diff
@@ -76,13 +79,14 @@ public int getTasksCount() {
      * @param description Description of the task.
      */
-    public void addToDoToList(Boolean isDone, String description) {
+    public Task addToDoToList(Boolean isDone, String description) {
```
Maybe the adding specific types of tasks could be handled by the subclasses of Task instead?

miljyy reviewed (commented):
Overall the code is easy to read and understand, I really like how clean everything is!

yongning0310 committed:
Added asserts statement to Parser and Storage.
In Parser, assert statement is added to ensure that event, deadline and todo are valid, it also ensures that output
is not null.
In Storage, assert statement is used to tasks of type T, D and E are valid when reading from file.

yongning0310 committed:
edited storage file

yongning0310 committed:
Merge pull request #1 from yongning0310/branch-A-Assertions

Added asserts statement to Parser and Storage.

yongning0310 committed:
refactored Duke, Parser and Storage, broke the logic into smaller parts
and more emphasis is placed on readability and maintainability with more descriptive names

yongning0310 committed:
Merge pull request #2 from yongning0310/branch-A-CodeQuality

refactored Duke, Parser and Storage, broke the logic into smaller parts

yongning0310 committed:
C-FlexibleDataSource

yongning0310 committed:
Merge pull request #3 from yongning0310/branch-BCD-Extension

C-FlexibleDataSource

yongning0310 committed:
created userguide and added Ui.png

yongning0310 committed:
fixed comments and styling

yongning0310 committed:
dependencies issue

yongning0310 committed:
jar file
